src/Microsoft.AspNetCore.Razor.Language/DefaultCodeDocumentFeature.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal class DefaultCodeDocumentFeature : RazorProjectEngineFeatureBase, IRazorCodeDocumentFeature
```

Comment:
interesting idea here. This would take a lot of what we do in tests with callbacks and things and turn it into extensibility.